trojsten/notifications/views.py

```diff
@@ -0,0 +1,43 @@
+from django.contrib.auth.decorators import login_required
+from django.http import HttpResponseForbidden, JsonResponse
+from django.shortcuts import get_object_or_404, redirect, reverse
+
+from .models import Notification
+
+
+@login_required
+def get_notifications(request):
+    notifications = (
+        Notification.objects.filter(user=request.user, was_read=False)
+        .order_by("-created_at")
+        .all()[:10]
+    )
+
+    serialized = []
+
+    for notification in notifications:
+        serialized.append(
+            {
+                "id": notification.pk,
+                "channel": notification.channel,
+                "message": notification.message,
+                "url": reverse("read_notification", args=(notification.pk,)),
+                "was_read": notification.was_read,
+                "created_at": notification.created_at,
+            }
+        )
+
+    return JsonResponse({"notifications": serialized, "unread": len(notifications)})
+
+
+@login_required
+def read_notification(request, notification_id):
+    notification = get_object_or_404(Notification, pk=notification_id)
+
+    if notification.user != request.user:
+        return HttpResponseForbidden()
+
+    notification.was_read = True
+    notification.save()
+
+    return redirect(notification.url)
```

Comment:
Tomuto redirectu uplne nerozumiem. Nechceme to proste vratit `HTTP 204`?

Reply:
ty sa na túto adresu dostaneš tak, že v maili / v zozname notifikácií klikneš na notifikáciu. toto nastaví, že si si ju pozrel a presmeruje ťa tam, kam si chcel ísť.

Reply:
Nechceme mat moznost "pozrel" bez kliknutia? Myslim ze tak to zvykne fungovat inde.